mk/main.mk

```diff
@@ -598,7 +598,7 @@ ALL_TARGET_RULES = $(foreach target,$(CFG_TARGET), \
 	$(foreach host,$(CFG_HOST), \
  all-target-$(target)-host-$(host)))
 
-all: $(ALL_TARGET_RULES) $(GENERATED) docs
+all: $(shell $(S)src/etc/generate_styles.py) $(ALL_TARGET_RULES) $(GENERATED) docs
```

Comment:
I believe `$(S)src/librustdoc/html/styles.rs` should have its own target, much like [how `llvmdeps.rs` used to be handled](https://github.com/rust-lang/rust/blob/2264049577e1d64a2a863a9b9f2b7d94f6b90e15/mk/llvm.mk#L60).  The appropriate stamp file (`stamp.rustdoc`, I think) [should depend on it](https://github.com/rust-lang/rust/blob/2264049577e1d64a2a863a9b9f2b7d94f6b90e15/mk/target.mk#L137).

I'm not sure if generating the file in `src` is appropriate or if it should be somewhere in the target output directory.  If the latter it should probably look like the [current `llvmdeps.rs` code](https://github.com/rust-lang/rust/blob/master/mk/llvm.mk#L82).  If it does end up in `src` it should be presumably be added to `.gitignore`.